src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -726,8 +724,13 @@ std::shared_ptr<MLModelWrapper> activity_classifier::export_to_coreml(
       {"type", "activity_classifier"},
       {"version", 2},
   };
+  for(const auto& kvp : additional_user_defined) {
+       user_defined_metadata.emplace_back(kvp.first, kvp.second);
+  }
+
   model_wrapper->add_metadata({
-      {"user_defined", std::move(user_defined_metadata)}
+      {"short_description", short_desc},
```

Comment:
could be moved. or we could have `const std::string& short_desc`. But to me, I get quite confused. The version should be anchored by the CPP side since all version info should come from CPP implementation. Otherwise, the python layer can pass the wrong version header back by the user, e.g., they are using 6.0 cpp backend but somehow 7.0 is passed from them. I bet user should not modify the `short_desc` on our python side.

But for me, I think it's better to provide a cpp backend version info. Even though we fast iterate the turicreate python implementation, the backend may not have that many changes and be stable for a long time. In which way it's more clear for user to identify backward compatibility.

Reply:
The short description gets displayed to the user when the open a CoreML model in Xcode. The should only know the Python version they used to make the model.